Improvements to matchNode

yungsters opened the pull request:
Makes a couple improvements to `matchNodes`:

- Allow `path.find(j.Identifier, node => WHITELIST.hasOwnProperty(node.name))`.
- Fix fatal in `path.find(j.Identifier, {loc: null})`;

Also, added unit tests.

cpojer commented:
thank you! Looks good. I'll let Felix merge and tag the new release, because he has the powers!

yungsters commented:
Updated this to support functions at any level:

```
matchNode(..., {
  type: 'MemberExpression',
  object: {
    type: 'Identifier',
    name: name => name === 'foo' || name === 'bar'
  },
});
```

Also, added unit tests.

fkling commented:
> Also, added unit tests.

:+1:
